frontend/packages/dev-console/src/components/gitops/GitOpsListPage.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import * as _ from 'lodash';
+import { PageHeading, LoadingBox } from '@console/internal/components/utils';
+import { NamespaceModel } from '@console/internal/models';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { userStateToProps } from '@console/internal/reducers/ui';
+import GitOpsList from './GitOpsList';
+import { GitOpsManifestData, GitOpsAppGroupData } from './gitops-types';
+
+interface StateProps {
+  user: K8sResourceKind;
+}
+
+const projectRes = { isList: true, kind: NamespaceModel.kind, optional: true };
+
+const GitOpsListPage: React.FC<StateProps> = ({ user }) => {
+  const [appGroups, setAppGroups] = React.useState(null);
+  const [emptyStateMsg, setEmptyStateMsg] = React.useState('');
+  const [namespaces, nsLoaded, nsError] = useK8sWatchResource<K8sResourceKind[]>(projectRes);
+  const userName = _.replace(user?.metadata?.name ?? '', /[^a-zA-Z0-9-]/g, '');
+  const annotation = 'app.openshift.io/vcs-uri';
+  const [secretNS, secretName] =
+    (userName && [`pipelines-${userName}-github`, `${userName}-github-token`]) || [];
+  const baseURL =
+    (secretNS &&
+      secretName &&
+      `/api/gitops/pipelines?secretNS=${secretNS}&secretName=${secretName}`) ||
+    undefined;
+
+  React.useEffect(() => {
```

Comment:
A lot of what's in this effect is useful to place in a separate utility which is easily testable on its own. This will simplify the component itself.